### PR TITLE
Make tray icon optional

### LINKED ISSUE #67: Upcoming version of Ubuntu (17.10) breaks ripme because of removal of tray
* Ripme version: 1.5.7, also tested 1.5.0
* Java version: openjdk version "1.8.0_144"
* Operating system: Ubuntu 17.10
* Exact URL you were trying to rip when the problem occurred: <at startup of ripme>
* Please include any additional information about how to reproduce the problem:

Simply start with `java -jar ripme.jar`

## Expected Behavior

GUI of ripme should open

## Actual Behavior

```
$ java -jar src/ripme.jar 
Loaded /home/emk2203/rip.properties
Loaded log4j.properties
Initialized ripme v1.5.7
Exception in thread "main" java.lang.UnsupportedOperationException
	at java.awt.TrayIcon.<init>(TrayIcon.java:144)
	at java.awt.TrayIcon.<init>(TrayIcon.java:168)
	at com.rarchives.ripme.ui.MainWindow.setupTrayIcon(MainWindow.java:947)
	at com.rarchives.ripme.ui.MainWindow.createUI(MainWindow.java:250)
	at com.rarchives.ripme.ui.MainWindow.<init>(MainWindow.java:164)
	at com.rarchives.ripme.App.main(App.java:51)

```
I suspect that the [reported tray icon removal in the new Ubuntu version](http://www.omgubuntu.co.uk/2017/09/will-you-miss-gnome-legacy-tray) is the problem here.

### PR DESCRIPTION
**Pull request by rautamiekka**
Maybe now I get along with GitHub ...

# Category

This change is exactly one of the following (please change `[ ]` to `[x]`) to indicate which:
* [x] a bug fix (Fix #67 https://github.com/4pr0n/ripme/issues/589)
* [ ] a new Ripper
* [ ] a refactoring
* [ ] a style change/fix

# Description

Please add details about your change here.
Just merely wrapped the function call in an empty `try...switch` statement, so it should work as is, and I can't test, yet.

# Testing

Required verification:
* [ ] I've verified that there are no regressions in `mvn test` (there are no new failures or errors).
* [ ] I've verified that this change works as intended.
  * [ ] Downloads all relevant content.
  * [ ] Downloads content from multiple pages (as necessary or appropriate).
  * [ ] Saves content at reasonable file names (e.g. page titles or content IDs) to help easily browse downloaded content.
* [ ] I've verified that this change did not break existing functionality (especially in the Ripper I modified).

Optional but recommended:
* [ ] I've added a unit test to cover my change.

### CODE REVIEW

**Comment by rautamiekka** [comment]
Finally ! Finally it looks like it should, but without having GitHub do it for me I couldn't get it work.

File: src/main/java/com/rarchives/ripme/ui/MainWindow.java [hunk 1]
```diff
@@ -247,7 +247,10 @@ public void run() {
     }
 
     private void createUI(Container pane) {
-        setupTrayIcon();
+        //If creating the tray icon fails, ignore it.
+        try {
+            setupTrayIcon();
+        } catch (Exception e) { }
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ui/MainWindow.java]: Yay pokemon exceptions ("Gotta `catch` em all!"). Ugly but effective! :P

Comment (rautamiekka) [reply src/main/java/com/rarchives/ripme/ui/MainWindow.java]: Pokemon didn't even come into mind when I coded this.

Totally agree it's ugly but is there any other way ? I were thinking it's better to abort the whole thing no matter what happens cuz it means the icon initialization didn't go flawlessly => the icon may not even work.

Comment (metaprime) [reply src/main/java/com/rarchives/ripme/ui/MainWindow.java]: Yeah, I think ignoring the exception in this case is the proper way to handle an exception here.

**Pull request merged**